include/clap/helpers/plugin.hxx

```diff
@@ -888,6 +896,44 @@ namespace clap { namespace helpers {
       return applyConfigurationSuccess;
    }
 
+   //----------------------//
+   // clap_plugin_surround //
+   //----------------------//
+   template <MisbehaviourHandler h, CheckingLevel l>
+   bool Plugin<h, l>::clapSurroundIsChannelMaskSupported(const clap_plugin_t *plugin,
+                                                         uint64_t channel_mask) noexcept {
+      auto &self = from(plugin);
+      auto methodName = "clap_plugin_surround.is_channel_mask_supported";
+      self.ensureMainThread(methodName);
+      self.ensureIsInactive(methodName);
+
+      return self.isChannelMaskSupported(channel_mask);
+   }
+
+   template <MisbehaviourHandler h, CheckingLevel l>
+   uint32_t Plugin<h, l>::clapSurroundGetChannelMap(const clap_plugin_t *plugin,
+                                                    bool is_input,
+                                                    uint32_t port_index,
+                                                    uint8_t *channel_map,
+                                                    uint32_t channel_map_capacity) noexcept {
+      auto &self = from(plugin);
+      auto methodName = "clap_plugin_surround.get_channel_map";
+      self.ensureMainThread(methodName);
+      self.ensureIsInactive(methodName);
```

Comment:
This is incorrect, we can query the channel map even if the plugin is active.

Reply:
Thanks for clarifying. Corrected via a2989d03b74c78bd28a493e9c9eba4c37335bb20